crates/core/src/node.rs

```diff
@@ -387,16 +387,21 @@ impl<'r, D: Doc> Node<'r, D> {
       .root
       .lang()
       .get_ts_language()
-      .field_id_for_name(name)
-      .unwrap_or(0);
+      .field_id_for_name(name);
     let root = self.root;
     let mut cursor = self.inner.walk();
     cursor.goto_first_child();
     let mut done = false;
+    
     std::iter::from_fn(move || {
+      if field_id.is_none() {
```

Comment:
I think this can be better, but I can change it later

Reply:
Yeah, I thought so too, but couldn't find a better way. The problem is that field id is now of type `NonZero` so `unwrap_or` wouldn't have worked. Perhaps I could early return a noop iter?

Reply:
I can clean it up later, no worry